bootstrap/main.go

```diff
@@ -111,6 +111,20 @@ func mainImpl() {
 		}
 	}
 
+	if !opts.GKE {
+		dns, err := kubectl.DeploymentAvailability(kubectlClient, "kube-dns", "kube-system")
+		if err != nil {
+			exitWithCapture("There was an error with getting DNS deployment status %s\n", err)
+		}
+
+		// We exit if the DNS pods are not up and running, as the installer needs to be
+		// able to connect to the server to correctly setup the needed resources.
+		if !dns {
+			fmt.Println("WARNING: DNS pods are not up and running, launcher requires correct DNS setup in the Kubernetes cluster.")
```

Comment:
We need to make this message sound as clear and as actionable as possible; we need the user to come away with the impression that they need to fix their cluster!
I'll try and think up some tweaks to the wording, will reply to this with ideas.

It's a little odd to say this is a warning if we then immediately exit!

Reply:
How about this:
```
Error: There is a problem with DNS resolution in your Kubernetes cluster.

The Weave Cloud launcher requires a working DNS resolver to continue with setup.
```
If we could add a simple test command to the end too, that'd be great!